go-controller/pkg/libovsdbops/db_object_ids.go

```diff
@@ -0,0 +1,213 @@
+package libovsdbops
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+const (
+	// ExternalIDs keys.
+	// OwnerControllerKey and OwnerTypeKey define managing entity (a.k.a. owner) for given db object.
+	// All the other ids are object-related.
+	// PrimaryIDKey will be used a primary client index.
+	// A combination of OwnerControllerKey, OwnerTypeKey, and ObjectNameKey will be used a secondary client index.
+	// While owner-related keys together with PrimaryIDKey will always be present in the ExternalIDs,
+	// ObjectNameKey may or may not be used, based on dbIDsMap.
+	OwnerControllerKey = types.OvnK8sPrefix + "/owner-controller"
+	OwnerTypeKey       = types.OvnK8sPrefix + "/owner-type"
+	// ObjectNameKey is a part of a secondary index, together with OwnerControllerKey and OwnerTypeKey
+	// May be used by controllers to store e.g. namespace+name of the object.
+	ObjectNameKey = types.OvnK8sPrefix + "/name"
+	// PrimaryIDKey will be used as a primary index, that is unique for every db object,
+	// and can be built based on the combination of all the other ids.
+	PrimaryIDKey = types.OvnK8sPrefix + "/id"
+)
+
+type dbObjType int
+
+type ownerType string
+
+// dbIDsMap is used to track indexes used by different objects.
+// ownerType is used directly in the ExternalIDs with OwnerTypeKey key.
+// The same ownerType may be used by different dbObjTypes,
+// therefore this map will make sure the same key is not used twice by any dbObjType.
+// To create a unique id, values for listed keys will be joined with colon in the order they are listed here,
+// to make unique ID easier to read, sort ids in the narrowing-scope order.
+var dbIDsMap = map[dbObjType]map[ownerType][]string{}
```

Comment:
doesn't this contain a pre-determined slice of valid string indexes? If so you should comment on that here.

Reply:
I am not sure I understand this comment, 
`dbIDsMap is used to track indexes used by different objects.` is supposed to say something around those lines. maybe adding pre-defined like `dbIDsMap is used to track pre-defined indexes used by different objects.` will help here?

Reply:
yeah

Reply:
it might even more clear to change the value to be an index, like:

```
type index string
var dbIDsMap = map[dbObjType]map[ownerType][]index{}
```

Reply:
so it ended up being `ExternalIDKey`